### ISSUE TITLE
Flashing as cursor is moved between windows on macOS

### ISSUE THREAD

**Issue by dae** [ankitects/anki#2263]
This appears to be a bug in Qt 6.4.x, triggered by the user of :hover.

https://forums.ankiweb.net/t/anki-2-1-55-beta-7-rc1/25130/17

@kleinerpirat it's a bit late to be switching Qt versions for this release, so I'm wondering if we can just disable some of the hovering as a partial/temporary workaround. Would it be easy to exclude the hovering on the deck list and top toolbar based on .isMac?

**Comment by kleinerpirat** [comment]
> I'm wondering if we can just disable some of the hovering as a partial/temporary workaround.

The question is whether disabling *some* of them fixes the issue? Is it just the main webview or are the other screens causing this too? I cannot test this, so I'm afraid you'll have to do it yourself..

> Would it be easy to exclude the hovering on the deck list and top toolbar based on .isMac?

Yes, I'd just use the [parent selector](https://sass-lang.com/documentation/style-rules/parent-selector) like this:

![image](https://user-images.githubusercontent.com/62722460/206883592-1d32d2d4-1531-4063-97bf-75844367769c.png)

As you can see, if an `&` comes after some selector, Sass will compile this into:

```sass
:not(.isMac) table:hover {
  @include elevation(2)
}
```

**Comment by dae** [comment]
The issue exists on any hover element, so it won't fully fix the problem, I was just thinking if we cover the most common cases that will at least reduce the issue. Any objections to me adding some :not(.isMac) lines to the scss files for the header/deck list? Or is there a more central way we could address it easily?

I've filed a report with Qt: https://bugreports.qt.io/browse/QTBUG-109333

**Comment by kleinerpirat** [comment]
> Or is there a more central way we could address it easily?

I'm afraid not. I just found this on StackOverflow:

```
*:hover {
    all: unset !important;
}
```
Credit: https://stackoverflow.com/a/28067492

... and thought we could try it on `deckbrowser.scss`. But it messes up everything big time, probably because it resets all hover styles of bootstrap and native browser styles, e.g. for buttons, too :sweat_smile:

**Comment by dae** [comment]
The issue appears to have existed since the switch to Qt 6.4, so I guess nobody got around to reporting it/had figured out reproduction steps yet. It seems to only happen when moving the cursor between windows.

I tried the :not(.isMac) approach, and while that disables the styling, it doesn't actually stop the flicker! It seems like I have to actually remove the css rule to stop the flicker. That's not going to be practical unfortunately.

I guess the remaining options are to release as-is, or to switch back to 6.3.2. I expect a 2.1.56 to follow within a few weeks of the .55 release, so I might hold off on the change for .55 and then fix it in .56. This'll be a soft release with no update notification, so it shouldn't affect a great number of users.

**Comment by kleinerpirat** [comment]
I guess we could risk opening other issues by reverting to Qt 6.3.2? Judging from the forum screencast, I think the flashing issue isn't such a big deal to justify A) removing all hover CSS rules or B) reverting the Qt version.

**Comment by dae** [comment]
I guess we'll find out with the release of .55 how many people pick up on it. :-) If we do end up reverting to 6.3, it would only be on Macs, as 6.4 seems ok on the other platforms, and brought Wayland improvements.

**Comment by glutanimate** [comment]
This can affect add-ons that create their own webviews quite severely, unfortunately, e.g.:

https://user-images.githubusercontent.com/5459332/207246625-377a5515-3296-441a-b6d7-f2075aec6edb.mp4

https://user-images.githubusercontent.com/5459332/207247281-f9e125ae-73dd-447a-b8dc-c9fce03608d2.mp4


It seems to get particularly bad when both web views are displaying dynamic content of some form, e.g. animations or cute cat GIFs. While the cute cat GIF scenario is not going to be that common, I do think the issue is general enough to were users will run into it to some degree, and since at its worst, the flashing can be *very* extreme, I would vote to revert to Qt 6.3.2 on Mac if we can (i.e. if there aren't any severe Qt issues that 6.4.0 addresses on Mac).

**Comment by glutanimate** [comment]
So to add onto the Qt bug discussion, my hunch would be that `:hover` elements are just one trigger of this bug. I would guess that any content that causes areas areas of a web view to be re-rendered could trigger this.

**Comment by dae** [comment]
Yeah, that's awful. I'll revert to Qt 6.3 for this release.

**Comment by dae** [comment]
Qt believe this will be fixed in 6.4.2, due out around mid Jan - will need to confirm.

**Comment by dae** [comment]
Still broken in 6.4.2. :-(
